src/validators/json_or_python.rs

```diff
@@ -0,0 +1,83 @@
+use pyo3::intern;
+use pyo3::prelude::*;
+use pyo3::types::PyDict;
+
+use crate::build_tools::SchemaDict;
+use crate::errors::ValResult;
+use crate::input::{Input, InputType};
+use crate::recursion_guard::RecursionGuard;
+
+use super::{build_validator, BuildContext, BuildValidator, CombinedValidator, Extra, Validator};
+
+#[derive(Debug, Clone)]
+pub struct JsonOrPython {
+    json: Box<CombinedValidator>,
+    python: Box<CombinedValidator>,
+    name: String,
+}
+
+impl BuildValidator for JsonOrPython {
+    const EXPECTED_TYPE: &'static str = "json-or-python";
+
+    fn build(
+        schema: &PyDict,
+        config: Option<&PyDict>,
+        build_context: &mut BuildContext<CombinedValidator>,
+    ) -> PyResult<CombinedValidator> {
+        let py = schema.py();
+        let json_schema: &PyDict = schema.get_as_req(intern!(py, "json_schema"))?;
+        let python_schema: &PyDict = schema.get_as_req(intern!(py, "python_schema"))?;
+
+        let json = build_validator(json_schema, config, build_context)?;
+        let python = build_validator(python_schema, config, build_context)?;
+
+        let name = format!(
+            "{}[json={}, python={}]",
```

Comment:
```suggestion
            "{}[json={},python={}]",
```
I think elsewhere we construct the name without spaces, might be wrong but let's be consistent.